packages/devextreme/js/__internal/ui/chat/messagelist.ts

```diff
@@ -169,6 +176,48 @@ class MessageList extends Widget<Properties> {
     });
   }
 
+  _shouldAddDayHeader(timestamp: undefined | string | number | Date): boolean {
+    const { showDayHeaders } = this.option();
+
+    if (timestamp === undefined || !showDayHeaders) {
+      return false;
+    }
+
+    const deserializedDate = dateSerialization.deserializeDate(timestamp);
+
+    if (!isDate(deserializedDate) || isNaN(deserializedDate.getTime())) {
+      return false;
+    }
+
+    return !dateUtils.sameDate(this._lastMessageDate, deserializedDate);
+  }
+
+  _createDayHeader(timestamp: string | number | Date | undefined): void {
+    const deserializedDate = dateSerialization.deserializeDate(timestamp);
+    const today = new Date();
+    const yesterday = new Date(new Date().setDate(today.getDate() - 1));
+    this._lastMessageDate = deserializedDate;
```

Comment:
We can’t create cache variables for each task. We have another cache—the group cache—that is already used to obtain information about the last rendered message in the previous group.

We should unify our approaches with the previous solution by either using the existing group cache or developing a more universal solution suitable for both cases.

It might look something like this:

```_shouldAddDayHeader(timestamp: undefined | string | number | Date): boolean {
    const { showDayHeaders } = this.option();

    if (!showDayHeaders) {
      return false;
    }

    const deserializedDate = dateSerialization.deserializeDate(timestamp);

    if (!isDate(deserializedDate) || isNaN(deserializedDate.getTime())) {
      return false;
    }

    const lastMessageGroup = this._messageGroups?.[this._messageGroups.length - 1];

    if (lastMessageGroup) {
      const { items } = lastMessageGroup.option();
      const lastMessageGroupItem = items[items.length - 1];

      return !dateUtils.sameDate(dateSerialization.deserializeDate(lastMessageGroupItem.timestamp), deserializedDate);
    }

    return true;
  }```

Reply:
I don't like the idea of messing last day header with message groups. And your suggested solution won't work if 1st message have a timestamp, 2nd message don't have it and we add 3rd that also have a timestamp with the same date as the first one

Reply:
I demonstrated the draft solution and idea, hoping that what you needed was a ready-made solution from me

Reply:
https://github.com/DevExpress/DevExtreme/pull/28160/commits/22ef938385f9753c5034331321e032425975ebcd

Reply:
after some discussion, previous solution was reverted